### PR TITLE
Normalize Filenames

### PR DESCRIPTION
**Pull request by davschneller**
With this PR, all filenames follow now (in the `src` folder):

* Headers always have `.h` (`.hpp` has been removed)
* All files follow `PascalCase`. I.e. starting with an uppercase letters, followed by a sequence of uppercase and lowercase letters.

All includes have been adjusted (and no other code, really). A CI test with a small Python script also checks these properties.

(the files in the `auto_tuning` folder will be taken care by a different PR)

### CODE REVIEW

File: src/DynamicRupture/Output/Builders/ReceiverBasedOutputBuilder.cpp [hunk 1]
```diff
@@ -1,17 +1,17 @@
-#include "DynamicRupture/Output/Builders/ReceiverBasedOutputBuilder.hpp"
-#include "Common/constants.hpp"
+#include "DynamicRupture/Output/Builders/ReceiverBasedOutputBuilder.h"
+#include "Common/Constants.h"
 #include "DynamicRupture/Misc.h"
-#include "DynamicRupture/Output/DataTypes.hpp"
-#include "DynamicRupture/Output/OutputAux.hpp"
+#include "DynamicRupture/Output/DataTypes.h"
+#include "DynamicRupture/Output/OutputAux.h"
```
Comment (vikaskurapati) [inline src/DynamicRupture/Output/Builders/ReceiverBasedOutputBuilder.cpp:5]: Same comment with the Aux.

**Review by vikaskurapati — APPROVED** [review]
Very minor comments and we can merge immediately after addressing them. LGTM!

File: .ci/tidy.sh [hunk 2]
```diff
@@ -28,37 +28,37 @@ format() {
...
+        src/Initializer/BasicTypedefs.h
         src/Initializer/Boundary.h
         src/Initializer/DynamicRupture.h
         src/Initializer/DeviceGraph.h
         src/Initializer/GlobalData.h
         src/Initializer/GlobalData.cpp
         src/Initializer/InitialFieldProjection.h
         src/Initializer/InitialFieldProjection.cpp
-        src/Initializer/InputAux.hpp
+        src/Initializer/InputAux.h
```
[the remark this answers is not in the document]
Comment (davschneller) [reply .ci/tidy.sh:50]: I'd suggest we leave it here for now—and/or discuss it again a bit in more detail (we'd also need to change e.g. the `DeviceAux` folders then)

**Comment by davschneller** [comment]
> Very minor comments and we can merge immediately after addressing them. LGTM! Does the clang-tidy fail because it is still using the master's configuration to perform its checks?

Thanks alot for the approve! The `clang-tidy` check failed because #1156 included the `Common/Filesystem.h` header, before it was capitalized.

**Pull request merged**